### PR TITLE
Add github workflow to generate python wheel artifacts

### PR DESCRIPTION
**Pull request by aahmed-se**
Add github workflow to build python wheel artifacts for osx

### CODE REVIEW

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
run cpp tests

File: .github/workflows/python-whl-osx.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+
+  build-wheel:
+    name: build-wheel
+    runs-on: macos-latest
```
Comment (sijie) [inline .github/workflows/python-whl-osx.yaml]: we need to generate python clients for multiple osx versions, not just the latest mac. How can we do that?

File: .github/workflows/python-whl-osx.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: Python 37 wheel osx
```
Comment (sijie) [inline .github/workflows/python-whl-osx.yaml]: how do we trigger this workflow for releases?

File: .github/workflows/python-whl-osx.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+on:
+  pull_request:
+    branches:
+      - master
```
Comment (sijie) [inline .github/workflows/python-whl-osx.yaml]: we need to run this on release tags, no?

File: .github/workflows/python-whl-osx.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          make _pulsar -j8
+          cd python
+          python3 setup.py bdist_wheel
+          cd ../..
+
+      - uses: actions/upload-artifact@master
+        name: upload wheel artifact
+        with:
+          name: python-wheel-artifacts
+          path: pulsar-client-cpp/python/dist
```
Comment (sijie) [inline .github/workflows/python-whl-osx.yaml]: where are these artifacts uploaded to?

File: .github/workflows/python-whl-osx.yaml (hunk 2 above)
Comment (aahmed-se) [reply .github/workflows/python-whl-osx.yaml]: for releases we can add additional event types
https://developer.github.com/v3/activity/events/types/#createevent

File: .github/workflows/python-whl-osx.yaml (hunk 3 above)
Comment (aahmed-se) [reply .github/workflows/python-whl-osx.yaml]: yes we can add that
https://developer.github.com/v3/activity/events/types/#createevent

File: .github/workflows/python-whl-osx.yaml (hunk 4 above)
Comment (aahmed-se) [reply .github/workflows/python-whl-osx.yaml]: In github it self it's in the job run
https://github.com/apache/pulsar/pull/5931/checks?check_run_id=362655254 top right hand corner.

File: .github/workflows/python-whl-osx.yaml (hunk 1 above)
Comment (aahmed-se) [reply .github/workflows/python-whl-osx.yaml]: github only supports osx 10.15, one option is to use azure pipelines which has osx10.14 also but better option would be to change the setup.py and cmake files to build for older versions of osx on 10.15, it's doable requires setting a few environments variables and custom linker flags.

**Comment by sijie** [comment]
@aahmed-se : I think you should have provided the instructions of building and publishing python clients using the Azure pipeline that you and @merlimat  used before. Because we need to publish macos clients for 2.4.2 and 2.5.0.

**Comment by aahmed-se** [comment]
@sijie Here is what I used in the past

https://github.com/aahmed-se/pulsar-ci-test/blob/master/azure-pipelines.yml

https://dev.azure.com/alahmedse/apache-pulsar/_build/results?buildId=90&view=artifacts&type=publishedArtifacts

we can't put it in apache since apache projects can't work with azure pipelines.

Also @merlimat already has a pr out to fix some issues that should be merged before we release python artifacts for mac it fixes issues relating to linking brew specific paths.

**Comment by aahmed-se** [comment]
run cpp tests

**Comment by aahmed-se** [comment]
run cpp tests

File: .github/workflows/python-whl-osx.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: Python 37 wheel osx
+on:
+  pull_request:
```
Comment (sijie) [inline .github/workflows/python-whl-osx.yaml]: I don't think we should upload the wheel files to PyPI on pull requests. It should be on releases.

**Pull request closed without merging**